sdk/servicebus/Azure.Messaging.ServiceBus/tests/SessionReceiverLiveTests.cs

```diff
@@ -157,44 +156,39 @@ public async Task Peek_IncrementsSequenceNmber(int messageCt)
             }
         }
 
-        //[Test]
-        //public async Task NextSession()
-        //{
-        //    var sender = new ServiceBusSenderClient(ConnString, SessionQueueName);
-        //    var messageCt = 10;
-        //    HashSet<string> sessions = new HashSet<string>() { "1", "2", "3" };
-
-        //    // send the messages
-        //    foreach (string session in sessions)
-        //    {
-        //        await sender.SendRangeAsync(GetMessages(messageCt, session));
-        //    }
-
-        //    // create receiver not scoped to a specific session
-        //    var sessionClient = new SessionReceiverClient(ConnString, SessionQueueName, ReceiveMode.PeekLock);
-        //    for (int i = 0; i < 30; i++)
-        //    {
-        //        IAsyncEnumerable<ServiceBusMessage> peekedMessages = sessionClient.PeekRangeBySequenceAsync(
-        //            fromSequenceNumber: 1,
-        //            maxMessages: 10);
-        //        //var ct = 0;
-
-        //        await foreach (ServiceBusMessage peekedMessage in peekedMessages)
-        //        {
-        //            //TestContext.Progress.WriteLine($"{++ct}-{peekedMessage.MessageId}-{peekedMessage.SessionId}");
-        //            Assert.AreEqual(sessionClient.SessionId, peekedMessage.SessionId);
-        //        }
-        //        string prevSession = sessionClient.SessionId;
-        //        sessions.Remove(prevSession);
-        //        TestContext.Progress.WriteLine($"{i}-{prevSession}");
-        //        await sessionClient.NextSessionAsync();
-
-        //        //Assert.AreNotEqual(prevSession, sessionClient.SessionId);
-
-
-        //        // verify that broker is alternating sessions
-        //    }
-        //    Assert.AreEqual(0, sessions.Count);
-        //}
+        [Test]
+        public async Task RoundRobinSessions()
+        {
+            var sender = new ServiceBusSenderClient(ConnString, SessionQueueName);
```

Comment:
Depending on whether my PR goes first or yours does, one of us will need to do a bit of restructuring for this.   😄